Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilder.cs

```diff
@@ -63,14 +64,32 @@ public BocColumnRenderer[] CreateColumnRenderers ()
 
       PrepareSorting(sortingDirections, sortingOrder);
 
-      var firstValueColumnRendered = false;
+      var renderedRowIcon = false;
       var bocColumnRenderers = new List<BocColumnRenderer>(_columnDefinitions.Count);
       var visibleColumnIndex = GetInitialVisibleColumnIndex();
+
+      var enableIconOnFirstPossibleColumn = _columnDefinitions.All(
+          cd =>
+          {
+            return cd is not IBocColumnDefinitionWithRowIconSupport { RowIconEnabled: RowIcon.Preferred };
+          });
```

Comment:
There's an alternative option which may be easier:

You could select min-value column from the list and then simply do a reference-equals in the for-loop. Or perhaps `order-by + first` might be even easier to understand. Yes, this will cause an extra loop, but the code is easier to grasp (I think, your thoughts?) and the 20 or so columns that are typically a maximum number used in the UI don't cost us that much CPU cycles that it benefits from more complex code.
```
var iconColumn;
if (EnableIcon)
  iconColumn = _columnDefinitions.OfType<IBocColumnDefinitionWithRowIconSupport>().Where(cd=> cd.RowIcon != Disabled).OrderByDescending(cd => cd.RowIcon).FirstOrDefault();
else 
  iconColumn = null;
```

and below:
`var showIcon = object.ReferenceEuqls(columnDefinition, iconColumn)`

Reply:
Yea the logic in here is quite complex. I don't know if it's better to keep the complexity and add a comment for how it works, to have the "better" performance or just bin it and go for the less complex one either. 

I like the current implementation but I can see how it's going to take double the time to understand compared to the easier implementation.

Reply:
I would like to go through the pro and cons as you see them to have a shared understanding of which solution we should keep. From your desciption, I see you favor the performance-optimized version over the less-complex one. Is my summary correct?

Reply:
Since I suggested the inital `is not ... {}` stuff. I generally don't think that its that bad to read. Yes you need to know about the feature and what what it does but after that it should be pretty clear.

If my memory of boolean logic is correct we can also convert the `All` into an `Any` like this:

```
!_columnDefinitions.Any(cd => cd is IBocColumnDefinitionWithRowIconSupport { RowIconEnabled: RowIcon.Preferred });
```

I think this is pretty readable as "There exists no `IBocColumnDefinitionWithRowIconSupport` where `RowIconEnabled` is set to `RowIcon.Preferred`. But that just another option :)

Reply:
@a-ctor to avoid misunderstandings: my suggestion refers to removing about 15 lines of logic from the implementation, not simply rewriting the line with the ALL-statement. Am not sure if you also meant the whole part or just this one line.

Reply:
I was referring to just the 5 lines you used as context for the comment. I mistakenly thought that the discussion is about the complexity of those lines and since I was the one who suggested the `is not` stuff I wanted to comment on that. Since its about something else - carry on without me :p